clang/test/CIR/CodeGen/attribute-annotate-multiple.cpp

```diff
@@ -0,0 +1,73 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=BEFORE
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir -mmlir --mlir-print-ir-after=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=AFTER
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM
+double *a __attribute__((annotate("withargs", "21", 12 )));
+int *b __attribute__((annotate("withargs", "21", 12 )));
+void *c __attribute__((annotate("noargvar")));
+void foo(int i) __attribute__((annotate("noargfunc"))) {
+}
+// redeclare with more annotate
+void foo(int i) __attribute__((annotate("withargfunc", "os", 23 )));
+void bar() __attribute__((annotate("withargfunc", "os", 22))) {
+}
+
+// BEFORE: module @{{.*}}attribute-annotate-multiple.cpp" attributes {cir.lang =
+
+// BEFORE: cir.global  external @a = #cir.ptr<null> : !cir.ptr<!cir.double>
+// BEFORE-SAME: [#cir.annotation<name = "withargs", args = ["21", 12 : i32]>]
+// BEFORE: cir.global  external @b = #cir.ptr<null> : !cir.ptr<!s32i>
+// BEFORE-SAME: [#cir.annotation<name = "withargs", args = ["21", 12 : i32]>]
+// BEFORE: cir.global  external @c = #cir.ptr<null> : !cir.ptr<!void>
+// BEFORE-SAME: [#cir.annotation<name = "noargvar", args = []>]
+
+// BEFORE: cir.func  @_Z3fooi(%arg0: !s32i) attributes {annotations =
+// BEFORE-SAME: [#cir.annotation<name = "noargfunc", args = []>,
+// BEFORE-SAME: #cir.annotation<name = "withargfunc", args = ["os", 23 : i32]>],
+// BEFORE: cir.func  @_Z3barv() attributes {annotations =
+// BEFORE-SAME: [#cir.annotation<name = "withargfunc", args = ["os", 22 : i32]>],
+
+
+// AFTER: module {{.*}}attribute-annotate-multiple.cpp" attributes
+// AFTER-SAME: {cir.global_annotations = [
+// AFTER-SAME: #cir<annotation value ["a", #cir.annotation<name = "withargs", args = ["21", 12 : i32]>]>,
+// AFTER-SAME: #cir<annotation value ["b", #cir.annotation<name = "withargs", args = ["21", 12 : i32]>]>,
+// AFTER-SAME: #cir<annotation value ["c", #cir.annotation<name = "noargvar", args = []>]>,
+// AFTER-SAME: #cir<annotation value ["_Z3fooi", #cir.annotation<name = "noargfunc", args = []>]>,
+// AFTER-SAME: #cir<annotation value ["_Z3fooi", #cir.annotation<name = "withargfunc", args = ["os", 23 : i32]>]>,
+// AFTER-SAME: #cir<annotation value ["_Z3barv", #cir.annotation<name = "withargfunc", args = ["os", 22 : i32]>]>],
```

Comment:
As mentioned in another comment, please refer to this examples to understand how this should be printed:
https://github.com/llvm/clangir/pull/804#discussion_r1737248576
https://github.com/llvm/clangir/pull/804#discussion_r1729631202 
https://github.com/llvm/clangir/pull/804#discussion_r1737256485